styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/AbstractRunnerHandler.java

```diff
@@ -64,19 +67,29 @@ public final void transitionInto(RunState state, EventRouter eventRouter) {
         if(!applicableFn.test(maybeExecutionDescription.orElseThrow())) {
           return;
         }
+        safeTransitionInto(state, eventRouter);
+        break;
 
+      case ERROR:
```

Comment:
Thinking out loud here an ERROR would mean that there is only an ERROR in styx, right?
Flyte errors would cause a terminated state instead of an error state.
What if we see multiple error events coming in?